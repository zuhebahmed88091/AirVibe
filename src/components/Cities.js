import PropTypes from 'prop-types';

const Cities = ({ city, aqi }) => {
  let weatherStatus = '';

  if (aqi === 1) {
    weatherStatus = 'Good';
  } else if (aqi === 2) {
    weatherStatus = 'Fair';
  } else if (aqi === 3) {
    weatherStatus = 'Moderate';
  } else if (aqi === 4) {
    weatherStatus = 'Poor';
  } else if (aqi === 5) {
    weatherStatus = 'Very Poor';
  }

  return (
    <div className="cities-weather">
      <h2>{city}</h2>
      <div className="air-quality">
        <span>Quality of air:</span>
        <span>{weatherStatus}</span>
      </div>
      <div className="quality-index">
        <span>Quality index:</span>
        <span>{aqi}</span>
      </div>
    </div>
  );
};
Cities.propTypes = {
  city: PropTypes.string.isRequired,
  aqi: PropTypes.number.isRequired,
};
export default Cities;
