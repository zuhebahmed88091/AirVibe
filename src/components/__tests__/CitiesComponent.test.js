import React from 'react';
import { render } from '@testing-library/react';
import Cities from '../Cities';

describe('Cities', () => {
  it('should render Cities component', () => {
    const city = 'London';
    const aqi = 1;
    render(<Cities city={city} aqi={aqi} />);
    expect(Cities).toMatchSnapshot();
  });
});
