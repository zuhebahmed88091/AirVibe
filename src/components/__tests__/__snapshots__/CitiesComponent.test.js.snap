// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities should render Cities component 1`] = `[Function]`;
